src/_blog/libp2p-webrtc-browser-to-server.md

```diff
@@ -0,0 +1,270 @@
+
+The DTLS handshake is performed using fingerprints contained in the Offer and Answer SDPs. After the handshake is complete, data is sent between peers using the SCTP (Stream Control Transmission Protocol) protocol, encrypting messages with DTLS over UDP or TCP.
+
+
+## WebRTC in libp2p
+
+<!--
+participantspacing 20
+entryspacing 0.75
+
+participant Browser
+participant Server
+box over Server: Generate TLS Certificate
+box over Server: Listen on UDP Port
+box over Browser: Create RTCPeerConnection
+box over Browser: Create Server's Answer SDP
+box over Browser: Create Offer SDP
+Browser->Server:STUN Binding Request
+box over Server: Create Browser's Offer SDP
+
+Server->(1)Browser:DTLS Handshake
+Browser->(1)Server:DTLS Handshake
+box over Browser,Server: Full DTLS Handshake is 3 round trips
+
+Server->(1)Browser:Libp2p Noise Handshake
+Browser->(1)Server:Libp2p Noise Handshake
+
+
+Browser<->Server:Multiplex Send/Receive Framed Data
+-->
+![](https://i.imgur.com/jF69zwh.png)
+
+Connecting to a server from a browser in the WebRTC implementation in libp2p has some similarities but differs in several ways.  Many of the features supported in the WebRTC standard, such as video, audio, and STUN and Turn servers, are not needed in libp2p.  The primary WebRTC component that libp2p leverages is the [RTCDataChannels(https://developer.mozilla.org/en-US/docs/Web/API/RTCDataChannel).
+
+### Server Setup
+To establish a connection with the browser, the server performs the following steps:
+
+1. Generates a self-signed TLS certificate.
+1. Listens on a UDP port for incoming STUN packets.
+1. Assembles the multiaddress of the server, which is either known upfront or discovered.
+
+### Browser Connection
+The browser initiates the connection by performing the following actions:
+
+1. Creates an [RTCPeerConnection](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection).
+1. Generates the server's Answer SDP using the components in the multiaddress.
+1. Modifies the SDP, or "munges" it, to include an auto-generated ufrag and password, as well as the server's IP and port.
+1. Creates an Offer SDP and modifies it with the same values.
+1. Sets the Offer and Answer SDP on the browser, which triggers the sending of STUN packets to the server.
+
+
+### Server Response
+The server responds by creating the browser's Offer SDP using the values in the STUN Binding Request.
+
+### DTLS Handshake
+The browser and server then engage in a DTLS handshake to open a DTLS connection that WebRTC can run SCTP on top of. A [Noise handshake](https://github.com/libp2p/specs/blob/master/noise/README.md) is initiated by the server using the fingerprints in the SDP as input to the [prologue data](https://noiseprotocol.org/noise.html#prologue), and completed by the browser over the Data Channel. This handshake authenticates the browser, although Noise is not used for the encryption of data. A total of six roundtrips are performed.
```

Comment:
The handshakes authenticates both parties right? The certificate is not necessarily tied to the server's identity, meaning both parties rely on this handshake to for authentication.

Reply:
That a good point.  I'll change wording from `authenticates the browser` to `authenticates the browser and server`.